src/Game.js

```diff
@@ -1,35 +1,29 @@
-import React, { Component, createRef } from "react";
-
-class Game extends Component {
-  constructor(props) {
-    super(props);
-    const { boardSizeX, boardSizeY } = props;
-    this.spritesPos = this.generateSpritesPos(boardSizeX, boardSizeY);
-    this.moves = 0; // please check readme
-    this.remainingSprites = boardSizeY;
-    this.boardRef = createRef();
-    // initially user is placed in center
-    this.state = {
-      userPos: {
-        x: Math.floor(boardSizeX / 2),
-        y: Math.floor(boardSizeY / 2)
-      },
-      hasFinished: false
-    };
-  }
-
-  componentDidMount() {
-    this.focusBoard();
-  }
-
-  focusBoard() {
-    const boardRef = this.boardRef.current;
-    if(boardRef) {
-      boardRef.focus();
+import React, { useState, useEffect, useCallback, useRef } from "react";
+
+function Game({ boardSizeX, boardSizeY }) {
+  const [spritesPos] = useState(generateSpritesPos(boardSizeX, boardSizeY));
```

Comment:
can you call it lazily? Read this for info https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Yes! I created a new commit for this